### PR TITLE
Replace inline pip install with pip install from requirements*.txt

### PR DESCRIPTION
**Pull request by jchen351**
### Description
Replace inline pip install with pip install from requirements*.txt

### Motivation and Context
so that CG can recognize

### Dependency

- [x] https://github.com/microsoft/onnxruntime/pull/21085

### CODE REVIEW

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -2810,7 +2814,7 @@ def main():
             run_subprocess([emsdk_file, "activate", emsdk_version], cwd=emsdk_dir)
 
         if args.enable_pybind and is_windows():
-            install_python_deps(args.numpy_version)
+            install_python_deps('requirements-pybind.txt',args.numpy_version)
```
Comment (snnn) [inline tools/ci_build/build.py:2817]: Sorry, it has to be "requirements.txt". You may put the file in a different folder, with this name.

File: tools/ci_build/requirements/requirements-pybind.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+setuptools
+wheel
+pytest
+numpy==1.24.0 ; python_version < '3.12'
+numpy==1.26.0 ; python_version >= '3.12'
```
Comment (snnn) [inline tools/ci_build/requirements/requirements-pybind.txt:5]: I have a PR(#21085) that updates Numpy to a newer version. Could you wait it? I think we have conflicts here.

File: tools/ci_build/build.py [hunk 3]
```diff
@@ -870,14 +870,21 @@ def update_submodules(source_dir):
...
 
-def install_python_deps(numpy_version=""):
-    dep_packages = ["setuptools", "wheel", "pytest"]
-    dep_packages.append(f"numpy=={numpy_version}" if numpy_version else "numpy>=1.16.6")
-    dep_packages.append("sympy>=1.10")
-    dep_packages.append("packaging")
-    dep_packages.append("cerberus")
-    dep_packages.append("psutil")
-    run_subprocess([sys.executable, "-m", "pip", "install", *dep_packages])
+def install_python_deps(requirements_file='requirements.txt',numpy_version=None):
```
Comment (snnn) [inline tools/ci_build/build.py:873]: Or maybe we can totally remove this function.

Comment (jchen351) [reply tools/ci_build/build.py:873]: Do we want to keep the NumPy version overwritten capability?  We can remove the function unless we don't want this capability.

Comment (snnn) [reply tools/ci_build/build.py:873]: I think we no longer need to.

Comment (snnn) [reply tools/ci_build/build.py:873]: We could just use the latest numpy

File: tools/ci_build/requirements/requirements-pybind.txt (hunk 2 above)
Comment (jchen351) [reply tools/ci_build/requirements/requirements-pybind.txt]: update to `numpy >= 1.19.0`, same as your PR.

File: tools/ci_build/requirements/requirements-pybind.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+setuptools
+wheel
+pytest
+numpy >= 1.19.0
```
Comment (snnn) [inline tools/ci_build/requirements/requirements-pybind.txt:4]: Why this line has spaces between the operator ">=", but the next line does not?

Comment (snnn) [reply tools/ci_build/requirements/requirements-pybind.txt:4]: Are the spaces optional?

File: tools/ci_build/requirements/requirements-pybind.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+setuptools
```
Comment (snnn) [inline tools/ci_build/requirements/requirements-pybind.txt:1]: Should be tools/ci_build/requirements/pybind/requirements.txt . The filename must be requirements.txt.

**Review by snnn — APPROVED** [review]

**Pull request merged**